src/components/ha-service-control.ts

```diff
@@ -449,7 +452,23 @@ export class HaServiceControl extends LitElement {
             @value-changed=${this._dataChanged}
           ></ha-yaml-editor>`
         : filteredFields?.map((dataField) => {
+            const selector = dataField?.selector ?? { text: undefined };
+            const type = Object.keys(selector)[0];
+            const enhancedSelector = [
+              "action",
+              "condition",
+              "trigger",
+            ].includes(type)
+              ? {
+                  [type]: {
+                    ...selector[type],
+                    path: [dataField.key],
+                  },
+                }
+              : selector;
```

Comment:
Not very happy about this but I don't know how to have a cleaner way to pass the path to each selector.